### PR TITLE
Wip rspec

### PR DESCRIPTION
**Pull request by chrisevett**
@rendicott can you look at this

Here's a summary of changes.
1. Actually fixed a bug where we were popping the machines off twice. 
2. Added unit tests for all of the major functionality. We still could use some integration tests with rundeck and redis, but I'm not super worried about it. I think I finally understand rspec and its the shit.

Stuff I still need to do:
1. documentation updates. Its basically wrong at this point
2. refactor test code and get rubocop passing on everything
3. Update rakefile to run lint /test/ build tasks
4. Add travis ci config to use docker compose
5. Add travis ci config to run tests

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,3 +16,4 @@ gem 'faker'
 gem 'systemd-journal'
 gem 'rubocop'
 gem 'sinatra-base'
+gem 'mock_redis'
```
Comment (rendicott) [inline Gemfile:19]: Why use mock_redis instead of just using a full implementation in the docker?

File: lib/PoolsClosed/jobs.rb [hunk 2]
```diff
@@ -45,12 +47,15 @@ def wait_for_rundeck(execution_id)
...
-        doc = Nokogiri::XML(response[1])
-        execution_node = doc.xpath('//execution/@status')
-        execution_status = execution_node[0].value
+        execution_status = parse_status(response) 
       end
       execution_status
     end
+    
+    def parse_status(xml)
+      binding.pry
```
Comment (rendicott) [inline lib/PoolsClosed/jobs.rb]: Did you mean to leave this breakpoint here?

File: lib/PoolsClosed/machines.rb [hunk 3]
```diff
@@ -10,6 +10,8 @@ class Machines
     RLSD     = 'poolsclosed_releasedmachines'.freeze
     ERR      = 'poolsclosed_lasterror'.freeze
 
+    attr_reader :jobs
+
     def initialize(cnf)
```
Comment (rendicott) [inline lib/PoolsClosed/machines.rb:15]: I don't follow what's going on in this function.

**Review by rendicott — APPROVED** [review]
Looks good. I like the tests. See single comments.

File: Gemfile (hunk 1 above)
Comment (chrisevett) [reply Gemfile:19]: I was only worried about unit testing for this iteration. When I write integration tests we'll use the real deal.

File: lib/PoolsClosed/jobs.rb (hunk 2 above)
Comment (chrisevett) [reply lib/PoolsClosed/jobs.rb]: pfft. 3am checkins.

File: lib/PoolsClosed/machines.rb (hunk 3 above)
Comment (chrisevett) [reply lib/PoolsClosed/machines.rb:15]: attr_reader makes the jobs var accessible to rspec (or whatever external class). Is that what you're asking?

Comment (rendicott) [reply lib/PoolsClosed/machines.rb:15]: I think I put my comment in the wrong place. I was talking about the 'initialize' function.

**Comment by chrisevett** [comment]
@rendicott it's customary to paste emojis when you approve a PR. I'm going to have to put my foot down on this one :100: :100: :100:

**Pull request merged**